mongo_connector/oplog_manager.py

```diff
@@ -419,41 +419,74 @@ def join(self):
         self.running = False
         threading.Thread.join(self)
 
-    def _pop_excluded_fields(self, doc):
+    @classmethod
+    def find_field(cls, field, doc):
+        """Find the field in the document which matches the given field.
+
+        The field may be in dot notation, eg "a.b.c". Yields a tuple
+        (path, field_value).
+        """
+        path = field.split('.')
+        try:
+            for key in path:
+                doc = doc[key]
+        except (KeyError, TypeError):
+            return
+        yield path, doc
+
+    @classmethod
+    def find_update_fields(cls, field, doc):
+        """Find the fields in the update document which match the given field.
+
+        Both the field and the top level keys in the doc may be in dot
+        notation, eg "a.b.c". Yields tuples (path, field_value).
+        """
+        if field in doc:
+            yield [field], doc[field]
+            return
+        for key in doc:
+            if len(key) > len(field):
+                # Handle case where field is 'a' and key is 'a.b'
+                if key.startswith(field) and key[len(field)] == '.':
+                    yield [key], doc[key]
+                    # Continue searching, there may be multiple matches.
+                    # For example, 'a' will match 'a.b' and 'a.c'.
+            elif len(key) < len(field):
+                # Handle case where field is 'a.b' and key is 'a'
+                if field.startswith(key) and field[len(key)] == '.':
+                    remaining = field[len(key) + 1:]
+                    match = list(cls.find_field(remaining, doc[key]))
+                    if match:
+                        path, value = match[0]
+                        path.insert(0, key)
+                        yield path, value
+                        return
```

Comment:
That wouldn't work quite right. It's nessecary to continue searching in a sub-document. The case this code is handling can be shown with this example:
`find_update_fields('a.b.c.d', {'a.b': {'c': {'d': 1}}})` 

In this call we'll match the `'a.b'` to the beginning of `'a.b.c.d'` and search the sub-document for `'c.d'`, eg `cls.find_field('c.d', doc['a.b'])`

The insert at the beginning of the list only happens once so I don't see it as a problem and string manipulation is unavoidable here.